src/dotnet/commands/dotnet-build/DotnetProjectBuilder.cs

```diff
@@ -147,19 +157,19 @@ protected override CompilationResult RunCompile(ProjectGraphNode projectNode)
             finally
             {
                 StampProjectWithSDKVersion(projectNode.ProjectContext);
-                _incrementalManager.CacheIncrementalState(projectNode);
+                _incrementalManager.CacheIncrementalState(projectNode, CreateRuntimeContexts(projectNode));
```

Comment:
I don't like piping the runtime context everywhere, just so that at some point the compiler IO manager receives them. It couples more code to the knowledge of runtime contexts than need be.

I'd remove them and let only the compiler IO manager create them (it has the graph node)